src/stream.h

```diff
@@ -15,8 +15,11 @@ typedef struct streamID {
 
 typedef struct stream {
     rax *rax;               /* The radix tree holding the stream. */
-    uint64_t length;        /* Number of elements inside this stream. */
+    uint64_t length;        /* Current number of elements inside this stream. */
     streamID last_id;       /* Zero if there are yet no items. */
+    streamID first_id;      /* The first non-tombstone entry, zero if empty. */
+    streamID xdel_max_id;   /* The maximal ID that was deleted. */
+    uint64_t offset;        /* All time number of elements in this stream. */
```

Comment:
maybe rename to `entries_added`
and the one in CG to `entries_consumed`

if yo do, also the OFFSET token in XGROUP SETID

Reply:
`entries_added` could make sense in place of offset, but `entries_consumed` is misleading. It does count the number of xreads, but it is initialized to the `offset/entries_added` opportunistically. WDUT?